realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -183,10 +184,10 @@ public static synchronized void init(Context context) {
             if (context == null) {
                 throw new IllegalArgumentException("Non-null context required.");
             }
-            BaseRealm.applicationContext = context.getApplicationContext();
-            RealmCore.loadLibrary(BaseRealm.applicationContext);
+            RealmCore.loadLibrary(context);
             defaultConfiguration = new RealmConfiguration.Builder().build();
             RealmLog.add(BuildConfig.DEBUG ? new AndroidLogger(Log.DEBUG) : new AndroidLogger(Log.WARN));
+            BaseRealm.applicationContext = context.getApplicationContext();
```

Comment:
But only way to solve that would be to re-introduce the `isInitialized` variable or use the protected `Builder(Context)`. I was kinda hoping to get rid of the constructor, but seems easier to use that for now.